libtransmission/torrent.cc

```diff
@@ -1678,14 +1678,9 @@ tr_torrent_metainfo const& tr_torrent::VerifyMediator::metainfo() const
     return tor_->metainfo_;
 }
 
-std::optional<std::string> tr_torrent::VerifyMediator::find_file(tr_file_index_t const file_index) const
+bool tr_torrent::VerifyMediator::check_piece(tr_piece_index_t const piece) const
 {
-    if (auto const found = tor_->find_file(file_index); found)
-    {
-        return std::string{ found->filename().sv() };
-    }
-
-    return {};
+    return tr_ioTestPiece(tor_, piece);
```

Comment:
As long as I understand currently verification process goes parallel to downloads/uploads, without interference. tr_ioTestPiece() makes calls to cache. But cache access is now serialized. While it's very unlikely one would find gigabytes of data in a relatively small cache. Code deduplication is important, but checking pieces during downloading seems very different compared to massive sequential check during complete verification process.

Reply:
> Actually, when I perform manual verification I do want transmission to read data from disk

That is what's going to happen even if I didn't revert this change, because the files of the torrent to be verified always get flushed from the cache before verifying begins.

> But cache access is now serialized. While it's very unlikely one would find gigabytes of data in a relatively small cache.

For the same reason, this is not a concern at all.

> The block size to read from disk is important for good performance (see https://trac.transmissionbt.com/ticket/3231#comment:2 (25.05.2010)). It should be not 16 KB, but rather 128 KB.

Again, not a concern.